flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -273,7 +282,21 @@ public Collection<Map<String, List<T>>> process(
                 new PriorityQueue<>(NFAState.COMPUTATION_STATE_COMPARATOR);
 
         for (ComputationState computationState : nfaState.getPartialMatches()) {
-            if (isStateTimedOut(computationState, timestamp)) {
+            String currentStateName = computationState.getCurrentStateName();
+            boolean isCurrentStateTimeout =
+                    windowTimes.containsKey(currentStateName)
+                            && isStateTimedOut(
+                                    computationState,
+                                    timestamp,
+                                    computationState.getPreviousTimestamp(),
+                                    windowTimes.get(currentStateName));
+            boolean isLastStateTimeout =
```

Comment:
The names `isCurrentStateTimeout ` and `isLastStateTimeout ` are still confusing.